graylog2-web-interface/src/components/streams/Stream.jsx

```diff
@@ -78,10 +78,19 @@ class Stream extends React.Component {
 
     this.state = {
       loading: false,
+      showStreamRuleForm: false,
     };
   }
 
-  _onDelete = (stream) => {
+  _closeStreamRuleForm = () => {
+    this.setState({ showStreamRuleForm: false });
+  };
+
+  _openStreamRuleForm = () => {
+    this.setState({ showStreamRuleForm: true });
+  };
+
+  _onDelete= (stream) => {
     // eslint-disable-next-line no-alert
     if (window.confirm('Do you really want to remove this stream?')) {
```

Comment:
Could we use `components/bootstrap/BootstrapModalConfirm.jsx` instead of `window.confirm`?